qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UserAccountSteps.java

```diff
@@ -139,11 +139,64 @@ public void enterDoctorDetails() {
 	@And("a user clicks on save account button")
 	public void clickOnSaveUserAccountButton() {
 		userAccountPage.saveUserAccount();
-		userAccountPage.waitForPage();
 	}
 
 	@Then("the system adds user account into the users table")
 	public void systemAddsUserAccount() {
 		assertTrue(textExists("Account Saved Successfully"));
 	}
+
+	@Then("the system confirms password meets the password criteria")
+	public void systemConfirmsPasswordMeetsPasswordCriteria() {
+		assertFalse(textExists("Validation errors found Failed to save account details"));
+	}
+
+	@And("a user enters person details in the user account form") 
+	public void enterPersonalDetails() {
+		userAccountPage.enterPersonalDetails("Clerk", "Data");
+		userAccountPage.selectGender();
+	}
+
+	@And("a user enters user account details in the user account form")
+	public void enterUserAccountDetails() {
+		userAccountPage.clickOnAddUserAccount();
+		userAccountPage.setUsername(CLERK);
+		userAccountPage.setUserPrivilegeLevel("Full");
+	}
+
+	@And("a user sets the password with only upper case letters")
+	public void setPasswordWithOnlyUpperCaseLetters() {
+		userAccountPage.setUserPassword("DATACLERK123", "DATACLERK123");
+	}
+
+	@And("a user sets the password with only digits")
+	public void setPasswordsWithOnlyDigits() {
+		userAccountPage.setUserPassword("123459876", "123459876");
+	}
+
+	@And("a user sets the password with only letters")
+	public void setPasswordsWithOnlyLetters() {
+		userAccountPage.setUserPassword("dataclerk", "dataclerk");
+	}
+
+	@And("a user sets the password which is lower than 8 characters")
+	public void setPasswordBelowTheMinimumLength() {
+		userAccountPage.setUserPassword("123Abc", "123Abc");
+	}
+
+	@And("a user enters password that meets the password criteria")
+        public void userEnterPasswordThatMeetsThePasswordCriteria() {
+                userAccountPage.setUserPassword("Dataclerk123", "Dataclerk123");
+        }
+
+	@Then("the system throws validation error on the password input field")
+	public void systemThrowsValidationErrorOnPasswordInputField() {
+		assertFalse(userAccountPage.containsText(SYSTEM_ALERT));
+	}
+
+	@Then("the system throws a validation error message")
+        public void systemThrowsAvalidationError() {
+                List<String> validationErrors = userAccountPage.getValidationErrors();
+                assertTrue(userAccountPage.isDataCorrect(validationErrors));
```

Comment:
Which error is being thrown by the system here?

Reply:
what this does is to assert that there is no error message which would fail otherwise if `Validation errors found Failed to save account details` is thrown!!

Reply:
Can we rather assert the exact error that thrown by the system?

Reply:
for some reason it doesn't pick up the assertion hence build failure -> https://github.com/openmrs/openmrs-contrib-qaframework/runs/5704939239?check_suite_focus=true

Reply:
That means the current suggestion is not asserting anything. Lets assert the exact error(Validation errors found Failed to save account details) is actually thrown by the system

Reply:
actually thats what i did in this commit https://github.com/openmrs/openmrs-contrib-qaframework/pull/388/commits/ae07b4e1028aba694e11cab2a45e1d94ad1b05fe but ts causing the build failures stated above

Reply:
As mentioned earlier, the method

```js
List<String> validationErrors = userAccountPage.getValidationErrors(); assertTrue(userAccountPage.isDataCorrect(validationErrors));
```

is not asserting anything that is why the test is passing. 

@mherman22 what massage error is expected to be thrown?

Reply:
ts Validation errors found Failed to save account details

Reply:
Then we can validate that error message is thrown when the password does not meet the password criteria.

Reply:
something like `assertTrue(textExists("Validation errors found Failed to save account details"));`?

Reply:
I think that should work.

Reply:
@mherman22 according to the log at https://github.com/openmrs/openmrs-contrib-qaframework/runs/5709520573?check_suite_focus=true#step:7:12317 seems to be suggest that the message error is thrown as expected. Could you verify this and on a user story that expects the text and figure the better way to validate the step.

Reply:
yes the error is thrown as expected but the text stops at that particular stage for some minutes

Reply:
Is it that the error message delays to be displayed that is why the test is failing or something else?

Reply:
no actually the error message is displayed in time but after display, it lags for some minutes. but it later breaks and jumps to another scenario

Reply:
> As mentioned earlier, the method
> 
> ```js
>  List<String> validationErrors = userAccountPage.getValidationErrors(); 
>  assertTrue(userAccountPage.isDataCorrect(validationErrors));
> ```
> 
> is not asserting anything here that is why the test is passing.
> 
> @mherman22 what massage error is expected to be thrown?

btw what i forgot to say is the error message is displayed here too

Reply:
Yes that logic does not assert that the expected error message is displayed that is why when used it passes the test.